source/TagRestrictions/TagsChecker.cs

```diff
@@ -122,25 +130,55 @@ private bool RequiredAnyCheck(HashSet<string> sourceTags, HashSet<string> target
             return false;
         }
 
-        private bool RequiredCheck(HashSet<string> sourceTags, HashSet<string> targetTags,
-            Func<string, IEnumerable<string>> requiredTags, ChassisLocations location = ChassisLocations.None)
+        private bool RequiredAnyCheck(HashSet<string> sourceTags, HashSet<string> targetTags,
+          Func<string, IEnumerable<string>> requiredTags, ChassisLocations location = ChassisLocations.None)
         {
             foreach (var tag in sourceTags)
-            foreach (var requiredTag in requiredTags(tag))
             {
-                if (targetTags.Contains(requiredTag)) continue;
+                var hasMetAnyRequiredTags = true;
+                foreach (var requiredTag in requiredTags(tag))
+                {
+                    hasMetAnyRequiredTags = false;
+                    if (targetTags.Contains(requiredTag))
+                    {
+                        hasMetAnyRequiredTags = true;
+                        break;
+                    }
+                }
+
+                if (hasMetAnyRequiredTags)
+                    continue;
 
                 var tagName = NameForTag(tag);
-                var requiredTagName = NameForTag(requiredTag);
+                var requiredTagNames = NameForTags(requiredTags(tag).ToHashSet());
```

Comment:
Don't like the conversion to HashSet here, either put it in `NameForTags` or is the HashSet even required?